public/scripts/sysprompt.js

```diff
@@ -0,0 +1,236 @@
+    $content.val(power_user.sysprompt.content);
+    if (!CSS.supports('field-sizing', 'content')) {
+        await resetScrollHeight($content);
+    }
+}
+
+/**
+ * Checks if the instruct template has a system prompt and prompts the user to save it as a system prompt.
+ * @param {string} name Name of the instruct template
+ * @param {object} template Instruct template object
+ */
+export async function checkForSystemPromptInInstructTemplate(name, template) {
+    if ('system_prompt' in template && name && template.system_prompt && !system_prompts.some(x => x.name === name)) {
+        const html = await renderTemplateAsync('migrateInstructPrompt', { prompt: template.system_prompt });
+        const confirm = await callGenericPopup(html, POPUP_TYPE.CONFIRM);
+        if (confirm) {
+            const prompt = { name: name, content: template.system_prompt };
+            const presetManager = getPresetManager('sysprompt');
+            await presetManager.savePreset(prompt.name, prompt);
+            toastr.success(`System prompt "${prompt.name}" has been saved.`);
+        } else {
+            toastr.info('System prompt has been discarded.');
+        }
+
+        delete template.system_prompt;
+    }
+}
+
+function toggleSystemPromptDisabledControls() {
+    $enabled.parent().find('i').toggleClass('toggleEnabled', !!power_user.sysprompt.enabled);
+    $contentBlock.toggleClass('disabled', !power_user.sysprompt.enabled);
+}
+
+function enableSystemPromptCallback() {
+    power_user.sysprompt.enabled = true;
+    $enabled.prop('checked', true);
+    toggleSystemPromptDisabledControls();
+    saveSettingsDebounced();
+    return '';
+}
+
+function disableSystemPromptCallback() {
+    power_user.sysprompt.enabled = false;
+    $enabled.prop('checked', false);
+    toggleSystemPromptDisabledControls();
+    saveSettingsDebounced();
+    return '';
+}
+
+function toggleSystemPromptCallback(_args, state) {
+    if (!state || typeof state !== 'string') {
+        return String(power_user.sysprompt.enabled);
+    }
+
+    const newState = isTrueBoolean(state);
+    newState ? enableSystemPromptCallback() : disableSystemPromptCallback();
+    return String(power_user.sysprompt.enabled);
+}
+
+function selectSystemPromptCallback(args, name) {
+    if (!power_user.sysprompt.enabled && !isTrueBoolean(args.forceGet)) {
+        return '';
+    }
+
+    if (!name) {
+        return power_user.sysprompt.name;
+    }
+
+    const quiet = isTrueBoolean(args?.quiet);
+    const instructNames = system_prompts.map(preset => preset.name);
+    const fuse = new Fuse(instructNames);
+    const result = fuse.search(name);
+
+    if (result.length === 0) {
+        !quiet && toastr.warning(`System prompt "${name}" not found`);
+        return '';
+    }
+
+    const foundName = result[0].item;
+    $select.val(foundName).trigger('input');
+    !quiet && toastr.success(`System prompt "${foundName}" selected`);
+    return foundName;
+}
+
+jQuery(function () {
```

Comment:
For several reasons (that I am not sure I remember all anymore) I prefer actual setup/init functions, than the generic jQuery trick to execute on page load.
Otherwise you'll just get race conditions with stuff not being set up again, or it running while the loading spinner is still there, etc.

I think I changed a few, and we should keep up the practice. Unless you have an argument against.

Just make an exported function and call that during init, similar to `initTags`, `initKeyboard`, `initPersonas` and all the other stuff.
We can control order better too.